pkg/chaosdaemon/tc_server.go

```diff
@@ -73,34 +73,18 @@ func generateQdiscArgs(action string, qdisc *pb.Qdisc) ([]string, error) {
 	return args, nil
 }
 
-func getAllInterfaces(ctx context.Context, log logr.Logger, pid uint32, enterNS bool) ([]string, error) {
-	processBuilder := bpm.DefaultProcessBuilder("ip", "-j", "addr", "show").SetContext(ctx)
-	if enterNS {
-		processBuilder = processBuilder.SetNS(pid, bpm.NetNS)
```

Comment:
It's getting interfaces in the remote network namespace. Using `net.Interfaces()` directly will break Chaos Mesh.

Reply:
We could use `net.Interfaces()` as a fallback, if `enterNS` is false.

Reply:
I tested `getAllInterfaces()`, it actually get the remote NICs.

Reply:
FYI <img width="859" alt="image" src="https://user-images.githubusercontent.com/43462394/180916487-0e0ae69a-6e44-4937-9ee2-9864d8ff6fc3.png">

Reply:
@FingerLeader I don't understand. You don't even tell him the target pid, how could it get interfaces in the network namespace of target process?

BTW, a normal single-stack container should only have `["lo", "eth0"]` :thinking: .

Reply:
I think that's because when deamon server calls the `SetTcs()`, the namespace has already been set to the target namespace.

Reply:
<img width="644" alt="image" src="https://user-images.githubusercontent.com/43462394/180930231-c955545c-59e0-4381-9430-dff2bd6a1809.png">
You can see when "handling tc request", `SetTcs`'s namespacedName is "default/ubuntu".

Reply:
> the namespace has already been set to the target namespace.

I'm pretty sure it doesn't work like this. The `chaos-daemon` never changes itself's net namespace. The interfaces of `chaos-daemon` and target net namespaces could be nearly the same (as it doesn't use host network), you cannot tell it's the target net namespace only by the interfaces.

Reply:
`chaos-daemon` spawns `nsexec` and `nsexec` will switch the net namespace and execute following commands (e.g. `tc`), so it never changes the net namespace of `chaos-daemon`.

Reply:
I checked ip of `eth0`, `net.Interfaces()` does get the wrong data 😢